Storage/SQL/SQLiteHistory.swift

```diff
@@ -82,18 +82,30 @@ public class SQLiteHistory {
 
     private var ignoredSchemes = ["about"]
 
-    required public init(db: BrowserDB) {
+    required public init?(db: BrowserDB, version: Int? = nil) {
         self.db = db
         self.favicons = FaviconsTable<Favicon>()
-        db.createOrUpdate(self.favicons)
+        if !db.createOrUpdate(self.favicons) {
+            return nil
+        }
 
         // BrowserTable exists only to perform create/update etc. operations -- it's not
         // a queryable thing that needs to stick around.
-        db.createOrUpdate(BrowserTable())
+        if !db.createOrUpdate(BrowserTable(version: version ?? BrowserTable.DefaultVersion)) {
+            return nil
+        }
     }
 }
 
 extension SQLiteHistory: BrowserHistory {
+    public func removeSiteFromTopSites(site: Site) -> Success {
+        if let url = site.url.asURL,
+           let host = url.host {
```

Comment:
We ought to normalize this, right?

Can we just add a `domain` property to `Site` that does the right thing?